include/crocoddyl/multibody/actions/contact-fwddyn.hxx

```diff
@@ -84,9 +84,8 @@ void DifferentialActionModelContactFwdDynamicsTpl<Scalar>::calc(
 #ifndef NDEBUG
   Eigen::FullPivLU<MatrixXs> Jc_lu(d->multibody.contacts->Jc);
 
-  if (Jc_lu.rank() < d->multibody.contacts->Jc.rows()) {
-    assert_pretty(JMinvJt_damping_ > Scalar(0.),
-                  "A damping factor is needed as the contact Jacobian is not full-rank");
+  if (Jc_lu.rank() < d->multibody.contacts->Jc.rows() && JMinvJt_damping_ == Scalar(0.)) {
+    throw_pretty("A damping factor is needed as the contact Jacobian is not full-rank");
```

Comment:
Thanks for the changes Wolfgang. Why did you change to throw_pretty here? If it is already inside NDEBUG, I think we can leave it as an assert, no?

Reply:
I noticed it was in a `#ifndef NDEBUG`, environment, so the change does no harm, does it?

Reply:
I was simply surprised that you changed to throw inside the `#ifndef NDEBUG` environment. since we are not worried about performance here. But this is of no importance, for me it is okay either way.

Reply:
The reason was that when searching for `assert_pretty`, in the preview window this "looked" like a case where it should be throw instead of assert [as it's not comparing full matrices] - and only on closer inspection the wider ifndef NDEBUG block was visible - this way it won't show up as a "possible" missed throw_pretty case.